api: Add `creatorId` field to assets

victorges opened the pull request:
<!-------------------------------------------------------------------------
 | Thanks for send a pull request! 🎉
 | First, please make sure you familiar with the contribution guidelines
 | https://github.com/livepeer/livepeer.studio/blob/master/CONTRIBUTING.md
 -------------------------------------------------------------------------->

**What does this pull request do? Explain your changes. (required)**

This adds the `creatorId` field to assets, which right now is an unverified value
provided by the caller. We store it in an object with a `type` and `value` for 
future-proofness.

**Specific updates (required)**
 - Add `creatorId` field to `asset` as an object
 - Add `creatorId` field to `new-asset-payload` as the same object or also just a string for ergonomy
 - Map the field to the asset object on creation

**How did you test each of these updates (required)**
`yarn test`

**Does this pull request close any open issues?**
Implements DAT-65

**Checklist**

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

victorges committed:
api/schema: Add creatorId field to asset

victorges committed:
api/schema: Create creatorId on new-asset-payload

victorges committed:
api/asset: Set the creatorId on new assets

victorges commented on packages/api/src/schema/schema.yaml:
```diff
@@ -1426,6 +1438,12 @@ components:
+            - $ref: "#/components/schemas/asset/properties/creatorId"
+            - type: string
+              description:
+                Developer-managed (unverified) creator ID of the asset
```
Wonder if we need this helper syntax [1] or if it would be better to have developers explicitly specify the `"unverified"` string just so it's on their face lol

[1] (this means that when creating an asset one could not only specify `creatorId: {type:unverified, value: foo}` but also just `creatorId: foo`)

0xcadams replied:
Yeah, I like the helper syntax. Especially since we only support `unverified` currently, it seems clunky to require that to be passed in.

victorges committed:
api: Add a test for the new field

victorges committed:
api/test: Fix nested test

0xcadams reviewed (approved):
🚢

victorges committed:
api/schema: Fix tests

Just found out that the type field is obligatory
and not inferred automatically once you specify
properties lol
